tofhir-engine/src/main/scala/io/tofhir/engine/data/read/FileDataSourceReader.scala

```diff
@@ -31,7 +31,15 @@ class FileDataSourceReader(spark: SparkSession) extends BaseDataSourceReader[Fil
    * @throws NotImplementedError      If the specified source format is not implemented.
    */
   override def read(mappingSource: FileSystemSource, sourceSettings:FileSystemSourceSettings, schema: Option[StructType], timeRange: Option[(LocalDateTime, LocalDateTime)], limit: Option[Int] = Option.empty,jobId: Option[String] = Option.empty): DataFrame = {
-    val finalPath = FileUtils.getPath(sourceSettings.dataFolderPath, mappingSource.path).toAbsolutePath.toString
+
+    // Do not add absolute path if it is a hadoop path
+    val finalPath = if (sourceSettings.dataFolderPath.startsWith("hdfs://")) {
+      sourceSettings.dataFolderPath + "/" + mappingSource.path
```

Comment:
Please use Paths to generate new paths.

Reply:
I refactored using the URI class for this case because the path contains a ":" character, which is illegal for Paths.